.github/actions/local-network/action.yaml

```diff
@@ -0,0 +1,99 @@
+---
+name: "Start local network"
+description: "Setup and start dashmate local network"
+inputs:
+  image_org:
+    description: "Docker image organization"
+    required: true
+runs:
+  using: composite
+  steps:
+    - name: Setup Node.JS
+      uses: ./.github/actions/nodejs
+
+    - name: Restore JS build artifacts
+      uses: strophy/actions-cache@opendal-update
+      with:
+        bucket: multi-runner-cache-x1xibo9c
+        root: actions-cache
+        path: build-js-artifacts-${{ github.sha }}.tar
+        key: build-js-artifacts/${{ github.sha }}
+
+    - name: Unpack JS build artifacts archive
+      shell: bash
+      run: tar -xf build-js-artifacts-${{ github.sha }}.tar
+
+    - name: Get dashmate fingerprint
+      id: dashmate-fingerprint
+      shell: bash
+      run: echo "sha=$(git log -1 --format="%h" -- packages/dashmate)" >> $GITHUB_OUTPUT
+
+    # TODO: Use upload artifacts action instead
+    - name: Restore local network data
+      id: local-network-data
+      uses: strophy/actions-cache/restore@opendal-update
+      with:
+        bucket: multi-runner-cache-x1xibo9c
+        root: local-network-data
+        path: |
+          /home/ubuntu/.dashmate
+          **/.env
+          dashmate_volumes_dump
+        key: local-network-volumes/${{ steps.dashmate-fingerprint.outputs.sha }}
+
+    - name: Restore dashmate volumes
+      run: ./scripts/dashmate/volumes/restore.sh
+      shell: bash
+      if: steps.local-network-data.outputs.cache-hit == 'true'
+
+    - name: Setup local network
+      run: |
+        set -ex
+
+        # create tenderdash blocks every 10s to speed up test suite
+        yarn dashmate config set --config=local platform.drive.tenderdash.consensus.createEmptyBlocksInterval "10s"
+
+        # collect drive logs for bench suite
+        yarn dashmate config set --config=local platform.drive.abci.logs.stdout.level "trace"
+
+        ./scripts/setup_local_network.sh
+        ./scripts/configure_test_suite.sh
+        ./scripts/configure_dotenv.sh
+      shell: bash
+      if: steps.local-network-data.outputs.cache-hit != 'true'
+
+    - name: Dump dashmate volumes
+      run: ./scripts/dashmate/volumes/dump.sh
+      shell: bash
+      if: steps.local-network-data.outputs.cache-hit != 'true'
+
+    - name: Save local network data
+      uses: strophy/actions-cache/save@opendal-update
+      with:
+        bucket: multi-runner-cache-x1xibo9c
+        root: local-network-data
+        path: |
+          /home/ubuntu/.dashmate
+          **/.env
+          dashmate_volumes_dump
+        key: local-network-volumes/${{ steps.dashmate-fingerprint.outputs.sha }}
+      if: steps.local-network-data.outputs.cache-hit != 'true'
+
+    - name: Configure pre-built docker images
```

Comment:
Where do we get these images  from? Looks like we assume they are already pushed (= we should have some dependency on another job declared above). Please add dependency/comment explaining that.

Reply:
It's not a job, it's an action that pulls images to do the work. Workflows what use this action defines dependencies to build-images job.